pkg/command/context.go

```diff
@@ -1239,3 +1274,126 @@ func prepareClusterServerURL(context *configtypes.Context, projectName, spaceNam
+	}
+
+	cfg, err := config.GetClientConfig()
+	if err != nil {
+		return nil, cobra.ShellCompDirectiveNoFileComp
+	}
+
+	var taeCtxs []*configtypes.Context
+	for _, ctx := range cfg.KnownContexts {
+		if ctx.Target == configtypes.TargetTAE {
+			taeCtxs = append(taeCtxs, ctx)
+		}
+	}
+	return completionFormatCtxs(taeCtxs), cobra.ShellCompDirectiveNoFileComp
+}
+
+func completeActiveContexts(_ *cobra.Command, args []string, _ string) ([]string, cobra.ShellCompDirective) {
+	if len(args) > 0 {
+		return nil, cobra.ShellCompDirectiveNoFileComp
+	}
+
+	currentCtxMap, err := config.GetAllCurrentContextsMap()
+	if err != nil {
+		return nil, cobra.ShellCompDirectiveNoFileComp
+	}
+
+	target := getTarget()
+
+	var allCtxs []*configtypes.Context
+	for _, ctx := range currentCtxMap {
+		if target == configtypes.TargetUnknown || target == ctx.Target {
+			allCtxs = append(allCtxs, ctx)
+		}
+	}
+	return completionFormatCtxs(allCtxs), cobra.ShellCompDirectiveNoFileComp
+}
+
+// Setup shell completion for the kube-context flag
+func completeKubeContext(_ *cobra.Command, _ []string, _ string) ([]string, cobra.ShellCompDirective) {
+	if kubeConfig == "" {
+		kubeConfig = getDefaultKubeconfigPath()
+	}
+
+	cobra.CompDebugln("About to get the different kube-contexts", false)
+
+	kubeclient, err := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
+		&clientcmd.ClientConfigLoadingRules{ExplicitPath: kubeConfig},
+		&clientcmd.ConfigOverrides{}).RawConfig()
+	if err != nil {
+		return nil, cobra.ShellCompDirectiveNoFileComp
+	}
+
+	var comps []string
+	for name, context := range kubeclient.Contexts {
+		comps = append(comps, fmt.Sprintf("%s\t%s@%s", name, context.AuthInfo, context.Cluster))
+	}
+	// Sort the completion to make testing easier
+	sort.Strings(comps)
+	return comps, cobra.ShellCompDirectiveNoFileComp
+}
+
+func completeCreateCtx(_ *cobra.Command, args []string, _ string) ([]string, cobra.ShellCompDirective) {
+	if len(args) == 0 {
+		comps := cobra.AppendActiveHelp(nil, "Please specify a name for the context")
+		return comps, cobra.ShellCompDirectiveNoFileComp
+	}
+
+	if endpoint == "" && kubeContext == "" {
+		// The user must provide more info by using flags.
+		// Note that those flags are not marked as mandatory
+		// because the prompt mechanism can be used instead.
+		comps := []string{"--"}
+		return comps, cobra.ShellCompDirectiveNoFileComp | cobra.ShellCompDirectiveNoSpace
+	}
+
+	// The user has provided enough information
+	return nil, cobra.ShellCompDirectiveNoFileComp
+}
+
+func completionFormatCtxs(ctxs []*configtypes.Context) []string {
+	var comps []string
+	for _, ctx := range ctxs {
+		info, _ := config.EndpointFromContext(ctx)
+
+		if info == "" && ctx.Target == configtypes.TargetK8s && ctx.ClusterOpts != nil {
+			info = fmt.Sprintf("%s:%s", ctx.ClusterOpts.Path, ctx.ClusterOpts.Context)
+		}
+
+		comps = append(comps, fmt.Sprintf("%s\t%s", ctx.Name, info))
+	}
+
+	// Sort the completion to make testing easier
```

Comment:
nit: I'd say from UX angle, having a predictable order also aids user scanning for the right item